### ISSUE TITLE
Custom scalar fields not serialized in stitched queries

### ISSUE THREAD

**Issue by nigel-sampson** [ChilliCream/hotchocolate#1220]
We've run into a regression when upgrading from `10.1.0` to `10.2.0`.  On investigating we've noticed that how variables are being serialized in upstream stitching queries has changed for custom scalars.

With `10.1.0` the variables on the request looked like the following, note that `startDate` and `endDate` are instances of a custom scalar `LocalDate`.
``` json
{
  "scopedContextData_organizationKey": "ffd6bd5b20a5b6f70b1b0553618e4dc5",
  "arguments_dateRange": {
    "startDate": "2019-11-11",
    "endDate": "2019-11-17",
    "timezoneId": "Pacific/Auckland"
  },
  "arguments_maxResults": null,
  "arguments_paging": {
    "offset": 0,
    "pageSize": 6
  }
}
```

With `10.2.0` the variables become the following:
```
{
  "scopedContextData_organizationKey": "ffd6bd5b20a5b6f70b1b0553618e4dc5",
  "arguments_dateRange": {
    "startDate": {
      "calendar": {
        "id": "ISO",
        "name": "ISO",
        "minYear": -9998,
        "maxYear": 9999,
        "eras": [
          {
            "name": "BCE"
          },
          {
            "name": "CE"
          }
        ]
      },
      "year": 2019,
      "month": 11,
      "day": 11,
      "dayOfWeek": 1,
      "yearOfEra": 2019,
      "era": {
        "name": "CE"
      },
      "dayOfYear": 315
    },
    "endDate": {
      "calendar": {
        "id": "ISO",
        "name": "ISO",
        "minYear": -9998,
        "maxYear": 9999,
        "eras": [
          {
            "name": "BCE"
          },
          {
            "name": "CE"
          }
        ]
      },
      "year": 2019,
      "month": 11,
      "day": 17,
      "dayOfWeek": 7,
      "yearOfEra": 2019,
      "era": {
        "name": "CE"
      },
      "dayOfYear": 321
    },
    "timezoneId": "Pacific/Auckland"
  },
  "arguments_maxResults": null,
  "arguments_paging": {
    "offset": 0,
    "pageSize": 6
  }
}
```

Debugging the custom scalar between versions shows that in `10.1.0` the custom scalar is never called, in `10.2.0` it is. 

So far my conclusion is that as part of #1134 the scalar is being deserialized as it comes into the service but not being re-serialized on the way back out.

**Comment by nigel-sampson** [comment]
Created a pull request with two failing tests #1221.

The issue doesn't seem to be just scalars as input values, they seem to work correctly, but when they're in delegated queries.

This should hopefully make it easier to narrow it down.

**Comment by michaelstaib** [comment]
@nigel-sampson awesome. Will do a fix and put it in 10.3.

**Comment by michaelstaib** [comment]
This one is now fixed and ships with 10.3 preview 8
